great_expectations/core/batch_manager.py

```diff
@@ -0,0 +1,169 @@
+import logging
+from collections import OrderedDict
+from typing import TYPE_CHECKING, Dict, List, Optional
+
+from great_expectations.core.batch import (
+    Batch,
+    BatchData,
+    BatchDefinition,
+    BatchMarkers,
+)
+from great_expectations.core.id_dict import BatchSpec
+
+if TYPE_CHECKING:
+    from great_expectations.execution_engine import ExecutionEngine
+
+logger = logging.getLogger(__name__)
+logging.captureWarnings(True)
+
+
+class BatchManager:
+    def __init__(
+        self,
+        execution_engine: ExecutionEngine,
+        batch_list: Optional[List[Batch]] = None,
+    ) -> None:
+        """
+        Args:
+            execution_engine: The ExecutionEngine to be used to access cache of loaded Batch objects.
+            batch_list: List of Batch objects available from external source (default is None).
+        """
+        self._execution_engine = execution_engine
+
+        self._active_batch_id = None
+        self._active_batch_data_id = None
+
+        if batch_list is None:
+            batch_list = []
+
+        self._batch_cache = {}
+        self._batch_data_cache = {}
+
+        self.load_batch_list(batch_list=batch_list)
+        if len(batch_list) > 1:
+            logger.info(
+                f"{len(batch_list)} batches will be added to this Validator.  The batch_identifiers for the active "
+                f"batch are {self.active_batch.batch_definition['batch_identifiers'].items()}"
+            )
+
+    @property
+    def execution_engine(self) -> ExecutionEngine:
```

Comment:
Can we delete this property? It seems like the `self._execution_engine` is an implementation detail of the batch manager and other objects don't need to reach inside here to get it?